vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -472,7 +483,7 @@ private void refreshValue() {
         JsonObject json = Json.createObject();
         json.put("key", keyMapper.key(value));
         dataGenerator.generateData(value, json);
-        setSelectedItem(json);
+        getElement().setPropertyJson("selectedItem", json);
```

Comment:
It seems there was no need to keep the `selectedItem` property around, so I inlined this usage.